### PR TITLE
Report only executable segmetents in Module events

### PR DESCRIPTION
**Pull request by dimitry-**
Since modules may end up being mapped between other modules,
and sometimes do (see http://b/191732187), we only report
now executable segment mappings for modules. This also brings
snapshot event data in line with module update data.

Since exectuable segment is located at the offset different
from 0 we need to take it into account while translating
absolute addresses to function address (and vice versa).
For this reason we introduce exectuable_segment_offset
to ModuleInfo

Bug: http://b/191732187
Test: Run orbit, intstrument a function, take a capture, go to
      Disassebly.

### CODE REVIEW

File: src/ObjectUtils/LinuxMap.cpp [hunk 1]
```diff
@@ -113,30 +102,16 @@ ErrorMessageOr<std::vector<ModuleInfo>> ParseMaps(std::string_view proc_maps_dat
...
-      address_range.is_executable |= is_executable;
-    }
-  }
-
-  std::vector<ModuleInfo> result;
-  for (const auto& [module_path, address_range] : address_map) {
-    // Filter out entries which are not executable
-    if (!address_range.is_executable) continue;
+    // Skip non-executable mappings
+    if (!is_executable) continue;
```
Comment (danielfenner) [inline src/ObjectUtils/LinuxMap.cpp:106]: This should do the fix for the case documented in b/191732187. I wonder if there are real world cases with multiple executable mappings per module. Probably there is little use in asking that - we'll find out ...

**Review by danielfenner — APPROVED** [review]
lgtm

File: src/ClientData/FunctionUtils.cpp [hunk 2]
```diff
@@ -63,7 +63,13 @@ std::optional<uint64_t> GetAbsoluteAddress(const orbit_client_protos::FunctionIn
 
   CHECK(!base_addresses.empty());
 
-  return func.address() + base_addresses.at(0) - module.load_bias();
+  auto result = func.address() + base_addresses.at(0) - module.load_bias() -
+                module.executable_segment_offset();
+  LOG("Calculating absolute address for %s: %#x(function_address) + %#x(base_address) - "
+      "%#x(load_bias) - %#x(segment_offset) = %#x",
+      func.pretty_name(), func.address(), base_addresses.at(0), module.load_bias(),
+      module.executable_segment_offset(), result);
```
Comment (dpallotti) [inline src/ClientData/FunctionUtils.cpp:71]: How often is this function called? Doesn't this spam the logs? This looks more intended for development.

File: src/ClientData/FunctionUtils.cpp [hunk 3]
```diff
@@ -63,7 +63,13 @@ std::optional<uint64_t> GetAbsoluteAddress(const orbit_client_protos::FunctionIn
 
   CHECK(!base_addresses.empty());
 
-  return func.address() + base_addresses.at(0) - module.load_bias();
+  auto result = func.address() + base_addresses.at(0) - module.load_bias() -
+                module.executable_segment_offset();
+  LOG("Calculating absolute address for %s: %#x(function_address) + %#x(base_address) - "
```
Comment (dpallotti) [inline src/ClientData/FunctionUtils.cpp:68]:
```suggestion
  LOG("Calculating absolute address for \"%s\": %#x(function_address) + %#x(base_address) - "
```

File: src/ObjectUtils/LinuxMap.cpp (hunk 1 above)
Comment (dimitry-) [reply src/ObjectUtils/LinuxMap.cpp:106]: yep - this is currently assumes that there is only one executable segment. I think for now it is good enough.

**Review by dpallotti — APPROVED** [review]
I'm just wondering about the `LOG`, looks good otherwise.

**Review by ronaldfw — APPROVED** [review]
LGTM

COFF file support should still work as nothing changes if executable segment offset is 0 (if I'm not mistaken), o/w I'll fix it

File: src/ClientData/FunctionUtils.cpp (hunk 2 above)
Comment (dimitry-) [reply src/ClientData/FunctionUtils.cpp:71]: ops - this was debugging leftover log - removed.

File: src/ClientData/FunctionUtils.cpp (hunk 3 above)
Comment (dimitry-) [reply src/ClientData/FunctionUtils.cpp:68]: removed.

**Pull request merged**